### PR TITLE
Prl 2940

### PR DESCRIPTION
**Pull request by SasankAchyuta9**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/index.ts [hunk 1]
```diff
@@ -143,7 +102,8 @@ const getStepsWithContent = (sequence: Step[], subDir = ''): StepWithContent[] =
   if (sequence?.length) {
     for (const step of sequence) {
       const { url } = parseUrl(step.url);
-      const stepDir = `${dir}${url.startsWith(subDir) ? url : `${subDir}${url}`}`;
+      const subdirurl = url.startsWith(subDir) ? url : `${subDir}${url}`;
```
Comment (vivek-sekhar) [inline src/main/steps/index.ts:105]: Why do we need subdir? what is the rationale behind it?

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/applicant/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts [hunk 2]
```diff
@@ -333,6 +320,83 @@ export const getRespondentDocuments = (sectionTitles, taskListItems, userCase, i
...
+    return {
+      id: 'medical_reports_respondent',
+      text: taskListItems.medical_reports_respondent,
+      href: url + URL.MEDICAL_REPORTS + '?byApplicant=No',
+    };
+  }
+};
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const isPaternityDocUploadedRespondent = (taskListItems, url, isPaternityDocUploaded): any => {
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts:379]: Are there test cases written to cover these individual methods?

File: src/main/steps/c100-rebuild/applicant/add-applicants/postController.ts [hunk 3]
```diff
@@ -40,35 +40,33 @@ export default class AddApplicantPostController extends PostController<AnyObject
...
-      (checkIfApplicantLengthLess && applicantFirstName === '') ||
-      (checkIfApplicantLengthLess && applicantLastName === '');
+      checkIfApplicantLengthLess && (applicantFirstName === '' || applicantLastName === '');
     const saveAndComeBackToggled = req.body['saveAndComeLater'];
 
     if (saveAndComeBackToggled) {
       req.session.save();
       return super.saveAndComeLater(req, res, req.session.userCase);
+    }
+    if (saveAndContinueChecked) {
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/applicant/add-applicants/postController.ts:50]: Are these changes tested and working as expected?

File: src/main/steps/c100-rebuild/child-details/childDetailPostController.ts [hunk 4]
```diff
@@ -76,7 +76,7 @@ export default class ChildDetailsPostController {
 
     req.session.userCase.cd_children = updatePartyDetails(
       {
-        ...getPartyDetails(childId, req.session.userCase.cd_children!),
+        ...getPartyDetails(childId, req.session.userCase.cd_children),
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/child-details/childDetailPostController.ts:79]: cd-children is an optional prop. Are you sure cd_children value does not come empty?

File: src/main/steps/common/dashboard-page.njk [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
 
 {% block pageTitle %}
-  {{ serviceName }} - {{ title }} - {{ govUk }}
+  {{ headingTitle }}{{ title }} - Private law - GOV.UK
```
Comment (vivek-sekhar) [inline src/main/steps/common/dashboard-page.njk:28]: Do we need both headingTitle and title on the dashboard page. if so, there is a hyphen missing between headingTitle & title.

File: src/main/steps/common/template.njk [hunk 6]
```diff
@@ -1,5 +1,8 @@
 {% extends "./page.njk" %}
 
+{% block pageTitle %}
+  {{ headingTitle }}{{ title }} - Private law - GOV.UK
```
Comment (vivek-sekhar) [inline src/main/steps/common/template.njk:4]: Hyphen missing. Is it not required?

File: src/main/steps/respondent/task-list/content.ts [hunk 7]
```diff
@@ -137,7 +137,7 @@ const en = () => ({
 });
 
 const cy = () => ({
-  title: '',
+  title: 'Respondent tasklist',
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/task-list/content.ts:140]: Temp welsh placeholder is missing.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/respondent/task-list/content.ts (hunk 7 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/task-list/content.ts:140]: added now vivek

File: src/main/steps/c100-rebuild/child-details/childDetailPostController.ts (hunk 4 above)
Comment (SasankAchyuta9) [reply src/main/steps/c100-rebuild/child-details/childDetailPostController.ts:79]: Lint is working fine on that.

File: src/main/steps/applicant/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts (hunk 2 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts]: test cases are written

File: src/main/steps/index.ts (hunk 1 above)
Comment (SasankAchyuta9) [reply src/main/steps/index.ts:105]: sonar lint is working fine on that.

File: src/main/steps/common/dashboard-page.njk (hunk 5 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/dashboard-page.njk:28]: added necessary changes

File: src/main/steps/common/template.njk (hunk 6 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/template.njk:4]: added necessary changes

File: src/main/steps/c100-rebuild/applicant/add-applicants/postController.ts (hunk 3 above)
Comment (SasankAchyuta9) [reply src/main/steps/c100-rebuild/applicant/add-applicants/postController.ts:50]: For a safer side this functionality is reverted back to original and also there are no complexity is found on this file.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/summary/utils.ts [hunk 8]
```diff
@@ -88,15 +88,18 @@ export const summaryList = (
   const summaryData: SummaryListRow[] = [];
   for (const key in keys) {
     const keyLabel = keys[key];
+    const getSelectedprivaedet = userCase[key] + getSelectedPrivateDetails(userCase);
+    const setkey = key1 => {
+      if (key1 === 'startAlternative' && userCase[key1] !== 'undefined') {
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/summary/utils.ts:93]: its from previous code.

File: src/main/steps/common/dashboard-page.njk [hunk 9]
```diff
@@ -24,8 +24,18 @@
 {% endblock %}
 
 
+{% set realtitle = ""%}
+{% if headingTitle and title === undefined %}
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/dashboard-page.njk:28]: done this with if and else.. and working fine.

**Pull request merged**